core-rust/state-manager/src/mempool/pending_transaction_result_cache.rs

```diff
@@ -366,7 +399,15 @@ impl PendingTransactionRecord {
             CheckMetadata::Cached => {
                 panic!("Precondition was not met - the result was cached, but the latest attempt was not a rejection")
             }
-            CheckMetadata::Fresh(StaticValidation::Valid(transaction)) => Ok(transaction),
+            CheckMetadata::Fresh(StaticValidation::Valid(transaction)) => {
+                Ok(PendingExecutedTransaction::new(
+                    transaction,
+                    self.latest_attempt
+                        .against_state
+                        .specific_version()
```

Comment:
Storing this isn't super accurate, because we lose a bit of specificity.

I think I'd prefer to store something like `ExecutedAtState::Committed { state_version }` or `ExecutedAtState::PendingPreparingVertices { base_committed_state_version }`, even if we use the "specific version" for the inverse index.

But it's not a big deal :D